src/lib/translators/mongo.ts

```diff
@@ -336,6 +524,27 @@ function transformEvolution(step: Readonly<S.EvolutionStep>): MongoStep {
   ];
 }
 
+/** transform a 'filter' step into corresponding mongo step */
+function transformFilterStep(step: Readonly<S.FilterStep>): MongoStep {
+  const condition = step.condition;
+  return { $match: buildMatchTree(condition) };
+}
+
+/** transform an 'ifthenelse' step into corresponding mongo step */
+function transformIfThenElseStep(
+  step: Readonly<Omit<S.IfThenElseStep, 'name' | 'newColumn'>>,
+): MongoStep {
+  const ifBlock: MongoStep = buildConditionTree(step.if);
+  const thenBlock = buildMongoFormulaTree(buildFormulaTree(step.then));
+  let elseBlock: MongoStep | string | number;
+  if (typeof step.else === 'string') {
+    elseBlock = buildMongoFormulaTree(buildFormulaTree(step.else));
+  } else {
+    elseBlock = transformIfThenElseStep(step.else);
+  }
+  return { $cond: [ifBlock, thenBlock, elseBlock] };
```

Comment:
:+1: 
Maybe it's easier to use the `{ $cond: { if: <boolean-expression>, then: <true-case>, else: <false-case> } }` syntax rather than the one that counts on the position inside the array. (To read more easily resulting queries)

Reply:
👍

Reply:
Fixed [here](https://github.com/ToucanToco/weaverbird/pull/510/files#diff-b4949f396e00c36a178d54c27a25dcf3R545)